packages/config/src/options.ts

```diff
@@ -191,12 +196,14 @@ const resolvedOptions: Array<ResolvedConfigOption> = [
     validation: validate.isBoolean,
     isExperimental: true,
     canUpdateDuringTestTime: false,
+    requireRestartOnChange: 'browser',
```

Comment:
this shouldn't require a full browser restart, since `window.fetch` gets set in the driver: https://github.com/cypress-io/cypress/blob/b0c8db34348be31f32795f3f67c3408403d59ace/packages/driver/src/cypress/cy.ts#L1024-L1030